frontend/src/concepts/pipelines/content/compareRuns/metricsSection/roc/utils.ts

```diff
@@ -0,0 +1,24 @@
+import { ROCCurveConfig } from '~/concepts/pipelines/content/artifacts/charts/ROCCurve';
+import { ConfidenceMetric } from './types';
+
+export const isConfidenceMetric = (obj: unknown): obj is ConfidenceMetric => {
+  const metric = obj as ConfidenceMetric;
+  return (
+    typeof metric.confidenceThreshold === 'number' &&
+    typeof metric.falsePositiveRate === 'number' &&
+    typeof metric.recall === 'number'
+  );
+};
+
+export const buildRocCurveConfig = (
+  confidenceMetricsArray: ConfidenceMetric[],
+  index: number,
+): ROCCurveConfig => ({
+  index,
+  data: confidenceMetricsArray.map((metric) => ({
+    name: metric.confidenceThreshold as unknown as string,
```

Comment:
Why cast a number type as a string? Shouldn't the number be converted to a string instead?

Reply:
this was part of a CP from KF. will fix